servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -337,16 +437,31 @@ public Completable closeAsyncGracefully() {
         return activeHosts.stream().map(Host::asEntry).collect(toList());
     }
 
-    private static final class Host<Addr, C extends ListenableAsyncCloseable> implements AsyncCloseable {
+    private static final class Host<Addr, C extends ListenableAsyncCloseable> implements ListenableAsyncCloseable {
         @SuppressWarnings("rawtypes")
         private static final AtomicReferenceFieldUpdater<Host, Object[]> connectionsUpdater =
                 AtomicReferenceFieldUpdater.newUpdater(Host.class, Object[].class, "connections");
 
         final Addr address;
+        private volatile boolean isExpired;
         private volatile Object[] connections = EMPTY_ARRAY;
 
+        private final ListenableAsyncCloseable closeable;
+
         Host(Addr address) {
             this.address = requireNonNull(address);
+            final AsyncCloseable ac = newCompositeCloseable().prepend(new AsyncCloseable() {
+                @Override
+                public Completable closeAsync() {
+                    return doClose(AsyncCloseable::closeAsync);
+                }
+
+                @Override
+                public Completable closeAsyncGracefully() {
+                    return doClose(AsyncCloseable::closeAsyncGracefully);
+                }
+            });
+            this.closeable = toListenableAsyncCloseable(ac);
```

Comment:
You can replace this whole thing with:
```java
this.closeable = toAsyncCloseable(graceful -> graceful ? doClose(AsyncCloseable::closeAsyncGracefully) :
        doClose(AsyncCloseable::closeAsync));
```
It will be consistent with how `asyncCloseable` is initialized at the end of RRLB ctor.